java/core/src/main/java/sleeper/core/table/TableId.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2022-2023 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package sleeper.core.table;
+
+import java.util.Objects;
+
+public class TableId {
```

Comment:
Is this the right name for this class, given it contains a field called `tableId`? It means that `TableIndex` has the method `Optional<TableId> getTableById(String tableId);` which looks wrong. Could call this class `TableIdAndName`? Alternatively rename the field `tableId` to `tableUniqueId` and then the method on `TableIndex` would be `Optional<TableId> getTableByUniqueId(String tableUniqueId);`?

Reply:
I've renamed the field, thanks.